backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -531,9 +569,9 @@ func initLogArchive() (logArchive archive.LogArchiveInterface) {
 }
 
 // NewClientManager creates and Init a new instance of ClientManager.
-func NewClientManager() ClientManager {
+func NewClientManager(onlyKubernetesWebhookMode bool) ClientManager {
```

Comment:
I think the flag is a package level global variable so it may be better to just access the flag value directly rather than a breaking change on a public method.

If you can't access it globally, perhaps an environment variable would be better.